src/backend/replication/walproposer.c

```diff
@@ -219,58 +225,75 @@ ShutdownConnection(int i)
 static void
 ResetConnection(int i)
 {
-	pgsocket sock; /* socket of the new connection */
-	WalKeeper *wk = &walkeeper[i];
+	pgsocket	sock;			/* socket of the new connection */
+	WalKeeper  *wk = &walkeeper[i];
 
 	if (wk->state != SS_OFFLINE)
 	{
 		ShutdownConnection(i);
 	}
 
-	/* Try to establish new connection
+	/*
+	 * Try to establish new connection
 	 *
 	 * If the connection information hasn't been filled out, we need to do
-	 * that here. */
+	 * that here.
+	 */
 	if (wk->conninfo[0] == '\0')
 	{
-		sprintf((char*) &wk->conninfo,
+		sprintf((char *) &wk->conninfo,
 				"host=%s port=%s dbname=replication options='-c ztimelineid=%s ztenantid=%s'",
 				wk->host, wk->port, zenith_timeline_walproposer, zenith_tenant_walproposer);
 	}
 
-	wk->conn = walprop_connect_start((char*) &wk->conninfo);
+	wk->conn = walprop_connect_start((char *) &wk->conninfo);
 
-	/* "If the result is null, then libpq has been unable to allocate a new PGconn structure" */
+	/*
+	 * "If the result is null, then libpq has been unable to allocate a new
+	 * PGconn structure"
+	 */
 	if (!wk->conn)
 		elog(FATAL, "failed to allocate new PGconn object");
 
-	/* PQconnectStart won't actually start connecting until we run PQconnectPoll. Before we do that
-	 * though, we need to check that it didn't immediately fail. */
+	/*
+	 * PQconnectStart won't actually start connecting until we run
+	 * PQconnectPoll. Before we do that though, we need to check that it
+	 * didn't immediately fail.
+	 */
 	if (walprop_status(wk->conn) == WP_CONNECTION_BAD)
 	{
-		/* According to libpq docs:
-		 *   "If the result is CONNECTION_BAD, the connection attempt has already failed, typically
-		 *    because of invalid connection parameters."
+		/*---
```

Comment:
Again, always wanted to ask what this `---` is intended to mean? I've seen it mostly in the file headers, but from time to time also in the bulk code comments like here

Reply:
The `---` tells pgindent to leave the comment alone. I added that manually, because without it, pgindend did this:
```
		/*
		 * According to libpq docs: "If the result is CONNECTION_BAD, the
		 * connection attempt has already failed, typically because of invalid
		 * connection parameters." We should report this failure.
		 *
		 * https://www.postgresql.org/docs/devel/libpq-connect.html#LIBPQ-PQCONNECTSTARTPARAMS
		 */
```
That's actually not too bad. But this block later in the file was mangled pretty badly:
```
+       /*
+        * PQflush returns: 0 if successful                    [we're good to move
+        * on] 1 if unable to send everything yet [call PQflush again] -1 if it
+        * failed                     [emit an error]
         */
```

Reply:
> The --- tells pgindent to leave the comment alone.

I see, good to know, thanks!